requests/adapters.py

```diff
@@ -284,7 +303,14 @@ def request_url(self, request, proxies):
         """
         proxy = select_proxy(request.url, proxies)
         scheme = urlparse(request.url).scheme
-        if proxy and scheme != 'https':
+        proxy_scheme = ''
+        if proxy:
+            proxy_scheme = urlparse(proxy).scheme
+
+        if proxy and proxy_scheme.lower().startswith('socks'):
```

Comment:
I see how you were being tricksy now. I wonder if we can be better about this.

We essentially have the same code path (`url = request.path_url`) if we're using a 'socks' proxy or if we're making an HTTPS request. What if we do

```py
proxied_http_request = proxy and scheme != 'https'
using_socks_proxy = False
if proxy:
    using_socks_proxy == urlparse(proxy).scheme.lower().startswith('socks')

url = request.path_url
if proxied_http_request and not using_socks_proxy:
    url = urldefragauth(request.url)
```

I puzzled through my logic with 

| proxy  | http_scheme && !socks_proxy_scheme | https_scheme     | socks_proxy_scheme |
|--------|------------------------------------|------------------|--------------------|
| truthy | urldefragauth(request.url)         | request.path_url | request.path_url   |
| falsey | request.path_url                   | request.path_url | request.path_url   |

where `http_scheme` is where `(scheme != 'https') is True` and `https_scheme` is where `(scheme != 'https') is False`. Hopefully the table makes sense.